plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml

```diff
@@ -66,26 +66,19 @@
     </library>
 
     <library
-          id="servlet-api-2.5"
-          name="Servlet API 2.5"
+          id="appengine-servlet-2.5"
+          name="App Engine: Servlet API 2.5"
           group="servlet"
           siteUri="http://www.oracle.com/technetwork/java/javaee/servlet/index.html"
           export="false">
+      <libraryDependency id="appengine-api"/>
```

Comment:
I had planned to get rid of inter-library dependencies now that we're using Maven. Are we still going to need this?

Reply:
The App Engine SDK is provided (hence here).  Prior to this change, `ServletClasspathContainer` determined the corresponding runtime and [explicitly listed the individual library jars to be included](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/2465/files#diff-046dccbf1bc4d935129f833a7579b5a5L101).  Now `ServletClasspathContainer` just includes a single `<library>` reference that corresponds to the Servlet API for the runtime.

Reply:
`<libraryDependency>` is useful to reference other `<library>` definitions, especially `appengine-api` since we have some nice package exclusion/inclusion rules to avoid surfacing repackaged/shaded types from `com.google.appengine.**.repackaged` and internal types from `com.google.apphosting`.